File: app/src/debug/java/com/kaciula/archiman/data/fake/FakeUsersDataSource.java
package com.kaciula.archiman.data.fake;

import com.kaciula.archiman.data.UserDataSource;
import com.kaciula.archiman.domain.entity.User;
import io.reactivex.Completable;
import io.reactivex.Observable;
import java.util.ArrayList;
import java.util.List;

public class FakeUsersDataSource implements UserDataSource {

  @Override
  public Observable<List<User>> getUsers() {
    List<User> users = new ArrayList<>(6);
    users.add(new User(1, "ka"));
    users.add(new User(2, "andrei"));
    users.add(new User(3, "arti"));
    users.add(new User(4, "marius"));
    users.add(new User(5, "stefan"));
    users.add(new User(6, "bogdan"));
    return Observable.just(users);
  }

  @Override
  public Completable createOrUpdateUsers(List<User> users) {
    return Completable.fromAction(() -> {
      // Do nothing here
    });
  }
}

File: app/src/debug/java/com/kaciula/archiman/data/fake/FakeUsersDataSource.kt
package com.kaciula.archiman.data.fake

import com.kaciula.archiman.data.UserDataStore
import com.kaciula.archiman.domain.entity.User
import io.reactivex.Completable
import io.reactivex.Observable
import java.util.*

class FakeUsersDataSource : UserDataStore {

    override fun getUsers(): Observable<List<User>> {
        val users = ArrayList<User>(6)
        users.add(User(1, "ka"))
        users.add(User(2, "andrei"))
        users.add(User(3, "arti"))
        users.add(User(4, "marius"))
        users.add(User(5, "stefan"))
        users.add(User(6, "bogdan"))
        return Observable.just<List<User>>(users)
    }

    override fun createOrUpdateUsers(users: List<User>): Completable {
        return Completable.fromAction {
            // Do nothing here
        }
    }
}
